### ISSUE TITLE
Add shell completions and manpage 

### ISSUE THREAD

**Issue by orhun** [lycheeverse/lychee#1564]
`lychee_bin` already uses `clap` which means we can utilize the following crates:

- [`clap_complete`](https://crates.io/crates/clap_complete): generated the shell completions
- [`clap_mangen`](https://crates.io/crates/clap_mangen): generated manpage

Then downstream sources (i.e. package managers) can consume the generated files :)

**Comment by thomas-zahner** [comment]
Yeah, that would definitely be two great improvements. This duplicates #1505

**Comment by mre** [comment]
Closing in favor of #1505. Will reverence the `clap` tools over there. Thanks!

**Comment by thomas-zahner** [comment]
I'm reopening this since I'm addressing the man page generation in  #1858 but not yet shell completions. This is why I renamed the issue from `Add shell completions and manpage` to `Add shell completions`.

Note that we should use the `--generate` flag to generate shell completions, probably in combination with [`clap_complete`](https://crates.io/crates/clap_complete).

**Comment by mre** [comment]
All done.

- Manpages were implemented in https://github.com/lycheeverse/lychee/pull/1858 and landed in 0.21.0.
- Shell completions were implemented in https://github.com/lycheeverse/lychee/pull/1972, landed in `master`, and will be released with lychee 0.24.0

Thanks, @orhun, for the idea. ✌️
